File: inaction-spring/inaction-spring-app/src/main/java/cn/orangepoet/inaction/spring/event/CustomSpringEventPublisher.java
package cn.orangepoet.inaction.spring.event;

import lombok.extern.slf4j.Slf4j;
import org.springframework.context.ApplicationEventPublisher;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class CustomSpringEventPublisher {
    private final ApplicationEventPublisher applicationEventPublisher;

    public CustomSpringEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
        this.applicationEventPublisher = applicationEventPublisher;
    }

    public void doStuffAndPublishEvent(String message) {
        CustomSpringEvent event = new CustomSpringEvent(this, message);
        applicationEventPublisher.publishEvent(event);
        log.info("threadId: " + Thread.currentThread().getId());
    }
}

File: inaction-spring/inaction-spring-app/src/main/java/cn/orangepoet/inaction/spring/event/CustomSpringEventPublisher.kt
package cn.orangepoet.inaction.spring.event

import org.slf4j.LoggerFactory
import org.springframework.context.ApplicationEventPublisher
import org.springframework.stereotype.Component

@Component
class CustomSpringEventPublisher(var applicationEventPublisher: ApplicationEventPublisher) {
    private val log = LoggerFactory.getLogger(CustomSpringEventPublisher::class.java)

    fun doStuffAndPublishEvent(message: String) {
        val event = CustomSpringEvent(this, message)
        applicationEventPublisher.publishEvent(event)
        log.info("threadId: " + Thread.currentThread().id)
    }
}